deploy-edpm.yml

```diff
@@ -37,20 +37,26 @@
   tags:
     - build-operators
 
+- name: Run pre_deploy hooks
```

Comment:
generally makes sense (and this is interesting https://docs.ansible.com/ansible/latest/collections/ansible/builtin/meta_module.html#ansible-builtin-meta-module-execute-ansible-actions)

not clear on the relevance of the pre_deploy hooks being added here though with the rest of this i.e. removing the whens and using end_play?

Reply:
pre_deploy hooks were duplicated in the sub-playbook, leading to duplicated imported playbook, with the same conditions being applied in order to avoid running them twice - but due to `import_*` pattern, it was running, with the `when` condition added to the content - more "skipping" than needed.

Reply:
So, we can't extract those hook runs: since adoption job is calling the `06-deploy-edpm.yml` directly, it has to run hooks from there:

https://github.com/rdo-infra/rdo-jobs/blob/13c683ea25eb0f2fd71412c5c8d305a674fd0540/playbooks/data_plane_adoption/setup_crc_use_install_yamls_extracted_crc.yaml#L48-L51

This explains why the adoption job was failing on this patch :).